lib/services/elasticsearch.js

```diff
@@ -153,718 +168,635 @@ class ElasticSearch extends Service {
-    esRequestSearch.scroll = '30s';
-
-    if (esRequestSearch.body.query === null) {
-      return Bluebird.reject(this.getError('document_id_cannot_be_null'));
-    }
-
-    return getAllIdsFromQuery(this.client, esRequestSearch)
-      .then(ids => {
-        return new Bluebird((resolve, reject) => {
-          esRequestBulk.body = ids.map(id => ({delete:
-            {_index: esRequestBulk.index,
-              _type: esRequestBulk.type,
-              _id: id}
-          }));
-
-          if (esRequestBulk.body.length === 0) {
-            return resolve({ids: []});
-          }
+      esIndex = this._getESIndex(index, collection),
+      esRequest = {
+        index: esIndex
+      };
 
-          return this.client.bulk(esRequestBulk)
-            .then(() => this.refreshIndexIfNeeded(esRequestBulk, {ids}))
-            .catch(error => reject(this.esWrapper.formatESError(error)));
-        });
+    return this.client.indices.getMapping(esRequest)
+      .then(({ body }) => {
+        const properties = includeKuzzleMeta
+          ? body[esIndex].mappings.properties
+          : _.omit(body[esIndex].mappings.properties, '_kuzzle_info');
+
+        return {
+          dynamic: body[esIndex].mappings.dynamic,
+          _meta: body[esIndex].mappings._meta,
+          properties
+        };
       });
   }
 
   /**
-   * Create an empty collection. Mapping will be applied if supplied.
+   * Adds a mapping definition to a specific type
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} mappings - Collection mappings in ES format
+   *
+   * @return {Promise.<Object>} { }
    */
-  createCollection(request) {
-    const esRequest = initESRequest(request);
-
-    return this.kuzzle.indexCache.exists(esRequest.index)
-      .then(indexExists => {
-        if (! indexExists) {
-          this.throw('index_does_not_exist');
-        }
-
-        return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type);
-      })
-      .then(collectionExists => {
-        if (collectionExists) {
-          return this.updateMapping(request);
-        }
-
-        const requestBody = request.input.body || {};
+  updateMapping (index, collection, mappings) {
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
 
-        this._checkMapping(requestBody);
+    return this.getMapping(index, collection, true)
+      .then(collectionMappings => {
+        this._checkMappings(collectionMappings);
 
-        esRequest.body = this._mergeDefaultMapping(
-          esRequest.index,
-          {},
-          requestBody
-        );
+        esRequest.body = _.merge(mappings, collectionMappings);
 
         return this.client.indices.putMapping(esRequest)
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
+          .then(() => {})
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this function is empty or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1186)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AEmptyBlock)